tests/unit/distribution/test_distribution.py

```diff
@@ -13,9 +13,134 @@
 from llama_stack.providers.datatypes import ProviderSpec
 
 
+class TestProviderMetrics:
+    """Test suite for provider metrics."""
+
+    @pytest.mark.asyncio
+    async def test_provider_with_valid_metrics_in_config(self):
+        """Test provider with valid metrics in config."""
+        run_config = StackRunConfig(
+            image_name="test_image",
+            providers={
+                "inference": [
+                    Provider(
+                        provider_id="test_provider",
+                        provider_type="test_type",
+                        config={"url": "http://localhost:8000", "metrics": "http://localhost:9090/metrics"},
+                    )
+                ]
+            },
+        )
+        provider_config = ProviderImplConfig(run_config=run_config)
+        provider_impl = ProviderImpl(provider_config, {})
+
+        response = await provider_impl.list_providers()
+        assert len(response.data) == 1
+
+        provider_info = response.data[0]
+        assert provider_info.provider_id == "test_provider"
+        assert provider_info.provider_type == "test_type"
+        assert provider_info.metrics == "http://localhost:9090/metrics"
+        assert "metrics" not in provider_info.config
+        assert provider_info.config["url"] == "http://localhost:8000"
+
+    @pytest.mark.asyncio
+    async def test_provider_with_invalid_metrics_in_config(self):
+        """Test invalid metrics in config fails when access /providers."""
+        run_config = StackRunConfig(
+            image_name="test_image",
+            providers={
+                "inference": [
+                    Provider(
+                        provider_id="test_provider",
+                        provider_type="test_type",
+                        config={"url": "http://localhost:8000", "metrics": "abcde-llama-stack"},
+                    )
+                ]
+            },
+        )
+        provider_config = ProviderImplConfig(run_config=run_config)
+        provider_impl = ProviderImpl(provider_config, {})
+
+        with pytest.raises(ValidationError):
+            await provider_impl.list_providers()
+
+    @pytest.mark.asyncio
+    async def test_provider_with_invalid_metrics_in_config2(self):
+        """Test invalid metrics2 in config fails when access /providers."""
+        run_config = StackRunConfig(
+            image_name="test_image",
+            providers={
+                "inference": [
+                    Provider(
+                        provider_id="test_provider",
+                        provider_type="test_type",
+                        config={"url": "http://localhost:8000", "metrics": 123},
+                    )
+                ]
+            },
+        )
+        provider_config = ProviderImplConfig(run_config=run_config)
+        provider_impl = ProviderImpl(provider_config, {})
+
+        with pytest.raises(ValidationError):
+            await provider_impl.list_providers()
```

Comment:
I'm wondering how relevant this test is, I mean, the previous one already validates the case with a malformed string. I don't think we want to assert for every single types.